h2o-core/src/main/java/water/parser/ParseDataset.java

```diff
@@ -412,11 +414,19 @@ public int compare(ParseWriter.ParseErr o1, ParseWriter.ParseErr o2) {
       String parseType = setup.getParseType().name();
       String[] originalColumnTypes = "PARQUET".equals(parseType) ? setup.getParquetColumnTypes() 
               : setup.getOrigColumnTypes();
-      if (originalColumnTypes != null) {
+      final int[] skippedColumns = setup.getSkippedColumns();
+      String[] newColumnTypes;
+      if (skippedColumns != null) {  // need to remove column types of skipped columns
+        List<Integer> skippedColIndices = Arrays.stream(skippedColumns).boxed().collect(Collectors.toList());
```

Comment:
I think a set should make the execution faster on a scenario with many skipped columns.
```suggestion
        Set<Integer> skippedColIndices = Arrays.stream(skippedColumns).boxed().collect(Collectors.toSet());
```